modules/git/repo_commit.go

```diff
@@ -211,9 +211,11 @@ func (repo *Repository) CommitsByFileAndRange(revision, file string, page int) (
 	}()
 	go func() {
 		stderr := strings.Builder{}
-		err := NewCommand(repo.Ctx, "rev-list", revision,
+		err := NewCommand(repo.Ctx, "rev-list",
 			"--max-count="+strconv.Itoa(setting.Git.CommitsRangeSize*page),
-			"--skip="+strconv.Itoa(skip), "--", file).
+			"--skip="+strconv.Itoa(skip),
+			"--end-of-options", revision,
```

Comment:
Instead of using the `--end-of-options` option, you could simply check if the `revision` has an initial `-` and if so prefix it with the `git.BranchPrefix`.

Reply:
but revision could be an sha or an full ref or an branch ...